Typescript compilation issues: TS2300: Duplicate identifier 'Promise'

Hey,

Feeling really hopeful about this library! Looks very similar to what I was about to embark on (so glad that I don't have to now!). I'm just having a little trouble getting up and running with it.

I've reduced my test project to the smallest possible case: Its only dependencies are coloquent itself and node typings, and the only file is an abstract `BaseModel` class that extends your `Model` class and adds some stuff. You clone the repo and try it yourself here https://github.com/kael-shipman/coloquent-test .

Attempting to compile this using typescript v3.2.2 results in a bunch of `TS2300: Duplicate identifier 'Promise'` errors:

```
node_modules/@types/es6-promise/index.d.ts:11:15 - error TS2300: Duplicate identifier 'Promise'.

11 declare class Promise<T> implements Thenable<T> {
                 ~~~~~~~

  ../../../../usr/lib/node_modules/typescript/lib/lib.es5.d.ts:1389:11
    1389 interface Promise<T> {
                   ~~~~~~~
    'Promise' was also declared here.
  ../../../../usr/lib/node_modules/typescript/lib/lib.es2015.iterable.d.ts:209:11
    209 interface Promise<T> { }
                  ~~~~~~~
    and here.
  ../../../../usr/lib/node_modules/typescript/lib/lib.es2015.promise.d.ts:216:13
    216 declare var Promise: PromiseConstructor;
                    ~~~~~~~
    and here.
  ../../../../usr/lib/node_modules/typescript/lib/lib.es2015.symbol.wellknown.d.ts:168:11
    168 interface Promise<T> {
                  ~~~~~~~
    and here.

node_modules/@types/es6-promise/index.d.ts:42:19 - error TS2300: Duplicate identifier 'Promise'.

42 declare namespace Promise {
                     ~~~~~~~

  ../../../../usr/lib/node_modules/typescript/lib/lib.es5.d.ts:1389:11
    1389 interface Promise<T> {
                   ~~~~~~~
    'Promise' was also declared here.
  ../../../../usr/lib/node_modules/typescript/lib/lib.es2015.iterable.d.ts:209:11
    209 interface Promise<T> { }
                  ~~~~~~~
    and here.
  ../../../../usr/lib/node_modules/typescript/lib/lib.es2015.promise.d.ts:216:13
    216 declare var Promise: PromiseConstructor;
                    ~~~~~~~
    and here.
  ../../../../usr/lib/node_modules/typescript/lib/lib.es2015.symbol.wellknown.d.ts:168:11
    168 interface Promise<T> {
                  ~~~~~~~
    and here.

.....
```

All of the errors seemed to originate in the `@types/es6-promise` package, so I tried manually removing that package from the `node_modules` hierarchy, but that then produced errors regarding not being able to find the name `Thenable`. As a final test, I restored the `@types/es6-promise` package, but deleted all type definitions other than `Thenable` and the project compiled.

I'm not quite enough of a typescript wizard to be able to fully debug this, but I have to think it may have to do with the version of typescript you're running, or possibly the version of node you're compiling against (or the one I'm compiling against, which is somewhat old because it has to remain in parity with an older production system for now).

Anyway, I'll dig around a bit and see if I can do something to fix it, then possibly submit a PR or update this issue.

Thanks again!

--- a/package.json
+++ b/package.json
@@ -50,6 +50,7 @@
     "chai-as-promised": "^7.1.1",
     "mocha": "^3.5.0",
     "moxios": "^0.4.0",
+    "typescript": "^2.8.1",
     "ts-node": "^3.3.0"
   }
 }

--- a/package.json
+++ b/package.json
@@ -50,7 +50,7 @@
     "chai-as-promised": "^7.1.1",
     "mocha": "^3.5.0",
     "moxios": "^0.4.0",
-    "typescript": "^2.8.1",
+    "typescript": "^3.0.0",
     "ts-node": "^3.3.0"
   }
 }

--- a/package.json
+++ b/package.json
@@ -38,7 +38,6 @@
     "dist": "npm run prepublish && npm publish"
   },
   "dependencies": {
-    "@types/es6-promise": "0.0.32",
     "@types/moxios": "^0.4.8",
     "axios": "^0.16.2",
     "php-date-formatter": "^1.3.4"
@@ -51,7 +50,6 @@
     "chai-as-promised": "^7.1.1",
     "mocha": "^3.5.0",
     "moxios": "^0.4.0",
-    "ts-node": "^3.3.0",
-    "typescript": "^2.8.1"
+    "ts-node": "^3.3.0"
   }
 }

--- a/src/httpclient/HttpClientPromise.ts
+++ b/src/httpclient/HttpClientPromise.ts
@@ -1,5 +1,6 @@
 import {AxiosResponse} from "axios";
 import {HttpClientResponse} from "./HttpClientResponse";
+import {Thenable} from "./Types";
 export interface HttpClientPromise
 {
     // /**

--- a/src/httpclient/Types.ts
+++ b/src/httpclient/Types.ts
@@ -1,0 +1,5 @@
+export interface Thenable<T> {
+    then<U>(onFulfilled?: (value: T) => U | Thenable<U>, onRejected?: (error: any) => U | Thenable<U>): Thenable<U>;
+    then<U>(onFulfilled?: (value: T) => U | Thenable<U>, onRejected?: (error: any) => void): Thenable<U>;
+}
+

--- a/src/httpclient/axios/AxiosHttpClientPromise.ts
+++ b/src/httpclient/axios/AxiosHttpClientPromise.ts
@@ -2,6 +2,7 @@ import {HttpClientResponse} from "../HttpClientResponse";
 import {AxiosPromise, AxiosResponse} from "axios";
 import {HttpClientPromise} from "../HttpClientPromise";
 import {AxiosHttpClientResponse} from "./AxiosHttpClientResponse";
+import {Thenable} from "../Types";
 
 export class AxiosHttpClientPromise implements HttpClientPromise
 {

